Tutorials/intro-to-QUAlibrate/calibrations/readme.md

```diff
@@ -0,0 +1,249 @@
+```python
+@node.run_action(skip_if=node.parameters.load_data_id is not None or not node.parameters.simulate)
+def simulate_qua_program(node):
+    qmm = QuantumMachinesManager(host=qop_ip, cluster_name=cluster_name)
+    samples, fig, wf_report = simulate_and_plot(qmm, config, node.namespace["qua_program"], node.parameters)
+    node.results["simulation"] = {
+        "figure": fig,
+        "wf_report": wf_report,
+        "samples": samples,
+    }
+
+```
+> 📥 This step connects to the OPX, simulates the program, shows the simulated output and stores results in node.results.
+
+
+### 6️⃣ Execute the Program
+
+Move your qm.execute(...) logic into a dedicated run action:
+
+```python
+@node.run_action(skip_if=node.parameters.load_data_id is not None or node.parameters.simulate)
+def execute_qua_program(node):
+    qmm = QuantumMachinesManager(host=qop_ip, cluster_name=cluster_name)
+    qm = qmm.open_qm(config)
+    job = qm.execute(node.namespace["qua_program"])
+    node.namespace["job"] = job
+
+    keys = [...]
+    data_fetcher = fetching_tool(job, data_list=keys, mode="wait_for_all")
+    values = data_fetcher.fetch_all()
+
+    for key, value in zip(keys, values):
+        node.results[key] = value
+```
+> 📥 This step connects to the OPX, runs the program, fetches the data, and stores results in node.results.
+
+
+### 7️⃣ Data Loading 
+If reusing saved results:
+```python 
+@node.run_action(skip_if=node.parameters.load_data_id is None)
+def load_data(node):
+     node.load_from_id(node.parameters.load_data_id)
+```
+> 📁 Useful for loading previously saved datasets during debugging or reanalysis.
+
+
+### 8️⃣ Data Analysis 
+Wrap your analysis logic inside a dedicated run action. 
+
+You can optionally move your processing and fitting functions to `calibration_utils/time_of_flight/analysis.py` for modularity and reuse — but you're also free to define them directly inside `analyse_data()` function.
+```python
+@node.run_action(skip_if=node.parameters.simulate)
+def analyse_data(node):
+    ...
+```
+> 🧠 This step processes the data.
+
+
+### 9️⃣ Data Plotting
+Wrap your plotting logic inside a dedicated run action. 
+As with the analysis step, you can optionally move your plotting functions into `calibration_utils/time_of_flight/plotting.py` for reuse and cleaner code organization. However, it’s equally valid to define plotting code directly inside `plot_data()` function.
+```python
+@node.run_action(skip_if=node.parameters.simulate)
+def plot_data(node):
+    ...
+```
+> 📊 These figures will show up in Python or in the QUAlibrate Web UI automatically.
+
+
+### 🔟 Save Results
+
+At the end of the node workflow, you should always include a @node.run_action() that calls node.save(). This ensures that the node’s parameters, results, figures, and metadata are properly stored and made available for later use.
+
+```python
+@node.run_action()
+def save_results(node):
+    node.save()
```

Comment:
Is this working as well, even though we don't have xarray datasets and quam?

Reply:
yes :)